hwangsae/transmuxer.c

```diff
@@ -215,31 +218,42 @@ hwangsae_transmuxer_parse_segments (HwangsaeTransmuxer * self,
 
   GList *segments = NULL;
   GList *it;
+  guint64 base_time_prev;
 
   for (; input_files; input_files = input_files->next) {
     gchar *file = input_files->data;
     Segment *segment;
-    guint64 segment_start;
+    guint64 segment_start, segment_end;
 
-    if (!hwangsae_common_parse_times_from_filename (file, &segment_start, NULL)) {
+    if (!hwangsae_common_parse_times_from_filename (file, &segment_start,
+            &segment_end)) {
       g_warning ("Invalid filename %s", file);
       continue;
     }
 
     segment = g_new0 (Segment, 1);
     segment->filename = g_strdup (file);
     segment->base_time = segment_start;
+    segment->end_time = segment_end;
 
     segments = g_list_insert_sorted (segments, segment, _compare_segments);
   }
 
+
+  base_time_prev = 0;
   /* Make base times start from zero. */
   for (it = g_list_last (segments); it; it = it->prev) {
+    if (base_time_prev != 0
```

Comment:
Hmm, for the sake of clarity I'd rather move the overlap check to `hwangsae_transmuxer_merge()`.